### ISSUE TITLE
new rule - typevar is only used once in a function signature

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3636]
```py
def bar[T](value: T) -> None: ...
```

[pyright reports the following error](https://basedpyright.com/?pyrightVersion=1.39.5&typeCheckingMode=all&reportUnusedParameter=false&code=CYUwZgBARghgTgbQCoF0AUA3GAbAriALgiQEoIBaAPggDkB7AO0IgDo2IBiYgTwAcQAavAgAiJCIgxe-eAGcIjbNwUMAxiAgBLBhADmIJnE2qIYXGoAumxhFmbdDGBdxwQQA):

```
TypeVar "T" appears only once in generic function signature
  Use "object" instead  (reportInvalidTypeVarUse)
```

this is more of a lint rule, because although it's not technically invalid, the typevar is useless.
